tests/functions_tests/test_point_to_point_communication.py

```diff
@@ -100,3 +102,31 @@ def test_communication(self):
             err = chainermn.functions.send(
                 y, self.communicator, self.rank_send)
             err.backward()
+
+    def test_retain(self):
+        if self.communicator.rank == 0:
```

Comment:
This test also works on more than 2 processes. It emulates `test_cycle_model` in `test_link.py`. FYI, we run test on 3 processes in the latest commit.